Account Onboard Utility/Accounts_Onboard_Utility.ps1

```diff
@@ -1292,284 +1494,311 @@ Write-LogMessage -Type Info -MSG "Getting PVWA Credentials to start Onboarding A
-								}
-								else
+								ElseIf ($sProp.Name -eq "platformAccountProperties")
 								{
-									# Update the existing account
-									$restBody = ConvertTo-Json $s_AccountBody -Depth 5 -Compress
-									$urlUpdateAccount = $URL_AccountsDetails -f $s_Account.id
-									$UpdateAccountResult = $(Invoke-Rest -Uri $urlUpdateAccount -Header $g_LogonHeader -Body $restBody -Command "PATCH")
-									if($null -ne $UpdateAccountResult)
+									ForEach ($sSubProp in $objAccount.platformAccountProperties.PSObject.Properties)
 									{
-										Write-LogMessage -Type Info -MSG "Account properties Updated Successfully"
-										$updateChange = $true
+										Write-LogMessage -Type Verbose -MSG "Updating Platform Account Properties $($sSubProp.Name) value to: '$($objAccount.platformAccountProperties.$($sSubProp.Name))'"
+										# Handle new Account Platform properties
+										$_bodyOp = "" | Select-Object "op", "path", "value"
+										$_bodyOp.op = "replace"
+										$_bodyOp.path = "/platformAccountProperties/" + $sProp.Name
+										$_bodyOp.value = $objAccount.platformAccountProperties.$($sProp.Name)
+										$s_AccountBody += $_bodyOp
 									}
 								}
+							}
+								
+							If ($s_AccountBody.count -eq 0)
+							{
+								Add-Content -Path "$CsvPath.propSkip" -Value ($account | ConvertTo-Csv -NoTypeInformation |Select-Object -Skip 1 ) 
+								Write-LogMessage -Type Info -MSG "No Account updates detected - Skipping"
+							}
+							else
+							{
+								# Update the existing account
+								$restBody = ConvertTo-Json $s_AccountBody -Depth 5 -Compress
+								$urlUpdateAccount = $URL_AccountsDetails -f $s_Account.id
+								$UpdateAccountResult = $(Invoke-Rest -Uri $urlUpdateAccount -Header $g_LogonHeader -Body $restBody -Command "PATCH")
+								if ($null -ne $UpdateAccountResult)
+								{
+									Add-Content -Path "$CsvPath.propUpdate" -Value ($account | ConvertTo-Csv -NoTypeInformation |Select-Object -Skip 1 ) 
+									Write-LogMessage -Type Info -MSG "Account properties Updated Successfully"
+									$updateChange = $true
+								}
+							}
 								
-								# Check if Secret update is needed
-								If(![string]::IsNullOrEmpty($objAccount.secret))
+							# Check if Secret update is needed
+							If (![string]::IsNullOrEmpty($objAccount.secret))
+							{
+								# Verify that the secret type is a Password (Only type that is currently supported to update
+								if ($objAccount.secretType -eq "password")
 								{
-									# Verify that the secret type is a Password (Only type that is currently supported to update
-									if($objAccount.secretType -eq "password")
+
+									$urlaccountPass = $URL_Password -f $s_Account.id
+									$body = "{`"reason`":`"Checking Password Sync with On-Prem`"`n`}"
+
+									$accountPassCheck = $(Invoke-Rest -Uri $urlaccountPass -Header $g_LogonHeader -Body $body -Command "Post")
```

Comment:
I don't think it is wise to get the password of a real account during this update
consider OTP, consider audit, consider Dual control
All these can result with workflows that are unwanted to the customer and does not add any value to the script
I think it can reduce our reliability in using this script since it will create the wrong impression of things